research-software-directory/RSD-as-a-service
# RSD admin can define software highlights

# RSD admin can define software highlights

Changes proposed in this pull request:
* Software highlights page is added to administration section of RSD
* RSD admin can add new highlights, change position of items (dnd) and delete items. See images.
* I added start and end date to highlights table. The idea is to define highlight item to be shown during specific time frame. This is just an idea and needs further discussion. This version does not support start-end date.

## Part of v2 release
**This PR will be merged into rsd-v2-release branch. This is long-standing branch to collect all features we want to release in v2. When starting to develop new feature for v2 use rsd-v2-release branch as your starting point.**

How to test:

* `make start` to rebuild app
* login as rsd_admin (professor3 if you have same .env definitions as I do)
* Navigate to software highlights page of admin section http://localhost/admin/software-highlights
* There should be some highlight items defined by data generation
* Add, delete or change position of highlight items.
* Software items already in the highlights are excluded from the search options

## Software highlights on admin page
![image](https://user-images.githubusercontent.com/9204081/225935727-eec601a6-6890-4576-a066-1925ef32e03f.png)

## Add software to highlights section
![image](https://user-images.githubusercontent.com/9204081/225935874-a779faba-0b76-49cb-83f1-f657c19890d6.png)

## Delete software from highlight section
![image](https://user-images.githubusercontent.com/9204081/225936080-e5bf6664-4d0a-4ed5-bdde-d23371f5f52b.png)

PR Checklist:
